mirgecom/steppers.py

```diff
@@ -80,18 +88,19 @@ def _advance_state_stepper_func(rhs, timestepper, checkpoint, get_timestep,
         if logmgr:
             logmgr.tick_before()
 
-        dt = get_timestep(state=state)
-        if dt < 0:
-            return istep, t, state
+        if get_timestep:
+            dt = get_timestep(state=state)
```

Comment:
`get_timestep` is mandatory, isn't it?

Reply:
No

Reply:
Ah, I missed the added `dt` parameter (it's not documented, BTW). I also don't really like that redundancy; may as well just do `advance_state(..., get_timestep=lambda _: dt)`.

Reply:
Sorry - I was scrambling to get to a meeting!  `get_timstep` is not mandatory because of constant dt, and the ability of the callbacks to compute dt.

Reply:
I'd prefer not to have multiple ways of doing the same thing; constant `dt` can be done by passing a constant function to `get_timestep`.

Reply:
> Ah, I missed the added `dt` parameter (it's not documented, BTW). I also don't really like that redundancy; may as well just do `advance_state(..., get_timestep=lambda _: dt)`.

I'm trying to make the minimum necessary changes to support #368, where the changes to this API were actually motivated, and discussed with @cmikida2.  I hope that PR will follow quickly.  Ultimately, it would be my preference to remove `get_timstep` altogether.

Reply:
> I'd prefer not to have multiple ways of doing the same thing; constant `dt` can be done by passing a constant function to `get_timestep`.

Can we fix this up in #368?  The current change set allows the current examples to work, doesn't break existing code, and supports changes from #368.  `get_timstep` is broken and #368 exposes that. I want to remove it, not promote it.

Reply:
Ah, so the pre-step callback is taking over the role of `get_timestep`? I haven't been following #368 closely. Why keep `get_timestep` around then? Can we deprecate it?

(Still need to document `dt`.)

Reply:
Yes, that is the intent, and what is working in #368.  Although I have not outright removed it in either one.  Since it is not a library function - it is a driver-based function, I am not sure it requires deprecation.  It's fully in user-space.

Reply:
> Ah, you mean deprecate the `get_timestep` argument to the stepper, I suppose?

👍

Reply:
Kinda wordy - but see how this grabs ya (6a7d20c).  I would feel better about deprecating it if the examples didn't use it any more - but it is OK in this instance I think because we'll fix it right away in the follow-on #368 where it introduces constant CFL mode.